native/cocos/renderer/pipeline/PlanarShadowQueue.cpp

```diff
@@ -88,15 +88,21 @@ void PlanarShadowQueue::gatherShadowPasses(scene::Camera *camera, gfx::CommandBu
             continue;
         }
 
-        if (!model->getInstanceAttributes().empty()) {
-            int i = 0;
-            for (const auto &subModel : model->getSubModels()) {
-                instancedBuffer->merge(model, subModel, i, subModel->getPlanarInstanceShader());
-                _instancedQueue->add(instancedBuffer);
-                ++i;
+        const auto &subModels = model->getSubModels();
+        const auto subModelCount = subModels.size();
+        for (uint32_t subModelIdx = 0; subModelIdx < subModelCount; ++subModelIdx) {
+            const auto &subModel = subModels[subModelIdx];
+            const auto &subModelPasses = subModel->getPasses();
+            for (index_t i = 0; i < static_cast<index_t>(subModelPasses.size()); ++i) {
```

Comment:
The same as above.

Reply:
merge function needs pass index so don't need change here